### PR TITLE
Support for long paths in run command

### PR DESCRIPTION
**Pull request by zateutsch**
`winapp run` fails on paths exceeding ~260 chars due to Windows MAX_PATH limits. This adds `longPathAware` manifest support, a `LongPathHelper` utility, and targeted workarounds for the two main failure points: Win32 build tools (MakePri/MakeAppx) and the WinRT `PackageManager` API.

## LongPathHelper

- `EnsureExtendedLengthPrefix` — adds `\\?\` (or `\\?\UNC\` for UNC paths) to bypass MAX_PATH in Win32 file I/O
- `GetShortPath` — converts long paths to 8.3 form via `GetShortPathName`; preserves trailing directory separators (critical for sparse registration URIs); consistent `<= MaxPath` boundaries throughout
- `GetShortPathOrThrow` — throws an actionable `InvalidOperationException` when shortening fails (8.3 disabled, path doesn't exist on disk)
- `ValidatePathLength` — fast-fails with a clear remediation message when long paths are unsupported at the system level
- `GetShortPathRaw` — correctly restores `\\?\UNC\server\share\...` → `\\server\share\...` (not the invalid `UNC\server\share\...`); handles `DllNotFoundException` gracefully on non-Windows

## RunCommand

- Validates `inputFolder.FullName` early (before the status spinner) so long-path errors surface before any file system work
- Also validates the resolved manifest path immediately after resolution

## PackageRegistrationService

- All three `PackageManager` call sites (`RegisterLooseLayoutAsync`, `RegisterSparseAsync`, `InstallPackageAsync`) use `GetShortPathOrThrow` instead of the silent `GetShortPath` fallback
- Sparse registration preserves the trailing separator on the external location URI

## PriService

- Uses the extended-prefix `dumpPath` consistently for `File.Exists`, `ReadAllTextAsync`, and `File.Delete` — not the unprefixed `dumpOutputFile`

## Tests

Added `LongPathHelperTests` with 16 unit tests covering boundary cases (`== MaxPath`, `> MaxPath`), UNC prefix handling, trailing separator preservation, and `GetShortPathOrThrow` throw behavior.

### CODE REVIEW

File: src/winapp-CLI/WinApp.Cli/Services/PackageRegistrationService.cs [hunk 1]
```diff
@@ -21,7 +22,12 @@ internal sealed class PackageRegistrationService(ILogger<PackageRegistrationServ
...
     public async Task RegisterLooseLayoutAsync(string manifestPath, CancellationToken cancellationToken = default)
     {
-        var manifestUri = new Uri(Path.GetFullPath(manifestPath));
+        var fullPath = Path.GetFullPath(manifestPath);
+        LongPathHelper.ValidatePathLength(fullPath);
+
+        // WinRT PackageManager doesn't support paths exceeding MAX_PATH or symlinks.
+        // Convert to 8.3 short path as a workaround.
+        var manifestUri = new Uri(LongPathHelper.GetShortPath(fullPath));
         var pm = new PackageManager();
```
[the remark this answers is not in the document]
Comment (zateutsch) [reply src/winapp-CLI/WinApp.Cli/Services/PackageRegistrationService.cs:31]: @copilot apply changes based on [this feedback](https://github.com/microsoft/winappCli/pull/398#discussion_r3041370498)

File: src/winapp-CLI/WinApp.Cli/Commands/RunCommand.cs [hunk 2]
```diff
@@ -187,6 +187,9 @@ public override async Task<int> InvokeAsync(ParseResult parseResult, Cancellatio
                     outputAppXDirectory ??= new DirectoryInfo(Path.Combine(inputFolder.FullName, "AppX"));
                     resolvedOutputDir = outputAppXDirectory;
 
+                    // Validate that the output path is usable (check long path support if needed)
+                    LongPathHelper.ValidatePathLength(outputAppXDirectory.FullName);
+
```
[the remark this answers is not in the document]
Comment (zateutsch) [reply src/winapp-CLI/WinApp.Cli/Commands/RunCommand.cs:245]: @copilot apply changes based on [this feedback](https://github.com/microsoft/winappCli/pull/398#discussion_r3041370513)

File: src/winapp-CLI/WinApp.Cli/Helpers/LongPathHelper.cs [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+        if (string.IsNullOrEmpty(directory))
+        {
+            return path;
+        }
+
+        var shortDir = GetShortPathRaw(directory);
+        var result = Path.Combine(shortDir, fileName);
+
+        return result.Length < MaxPath ? result : path;
+    }
```
[the remark this answers is not in the document]
Comment (zateutsch) [reply src/winapp-CLI/WinApp.Cli/Helpers/LongPathHelper.cs:116]: @copilot apply changes based on [this feedback](https://github.com/microsoft/winappCli/pull/398#discussion_r3041370541)

File: src/winapp-CLI/WinApp.Cli/Helpers/LongPathHelper.cs [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+
+        if (!IsSystemLongPathEnabled())
+        {
+            throw new InvalidOperationException(
+                $"The path exceeds the Windows MAX_PATH limit of {MaxPath} characters and long path support is not enabled on this system. " +
+                "To fix this, either:\n" +
+                "  1. Enable long paths: run 'reg add HKLM\\SYSTEM\\CurrentControlSet\\Control\\FileSystem /v LongPathsEnabled /t REG_DWORD /d 1 /f' as Administrator and restart your terminal, or\n" +
+                "  2. Use a shorter directory path.");
+        }
+    }
```
[the remark this answers is not in the document]
Comment (zateutsch) [reply src/winapp-CLI/WinApp.Cli/Helpers/LongPathHelper.cs:53]: @copilot apply changes based on [this feedback](https://github.com/microsoft/winappCli/pull/398#discussion_r3041370554)

File: src/winapp-CLI/WinApp.Cli/Helpers/LongPathHelper.cs [hunk 5]
```diff
@@ -0,0 +1,199 @@
...
+        {
+            return;
+        }
+
+        if (!IsSystemLongPathEnabled())
+        {
+            throw new InvalidOperationException(
+                $"The path exceeds the Windows MAX_PATH limit of {MaxPath} characters and long path support is not enabled on this system. " +
+                "To fix this, either:\n" +
+                "  1. Enable long paths: run 'reg add HKLM\\SYSTEM\\CurrentControlSet\\Control\\FileSystem /v LongPathsEnabled /t REG_DWORD /d 1 /f' as Administrator and restart your terminal, or\n" +
```
Comment (nmetulev) [inline src/winapp-CLI/WinApp.Cli/Helpers/LongPathHelper.cs]: Instead of adding the command here, should we instead redirect to the docs like https://learn.microsoft.com/en-us/windows/win32/fileio/maximum-file-path-limitation?tabs=registry#enable-long-paths-in-windows-10-version-1607-and-later (maybe we use an aka.ms)?

Comment (zateutsch) [inline src/winapp-CLI/WinApp.Cli/Helpers/LongPathHelper.cs]:
```suggestion
                "Visit https://aka.ms/enable-long-paths-on-windows for guidance on enabling long paths.
```

File: src/winapp-CLI/WinApp.Cli/Helpers/LongPathHelper.cs [hunk 6]
```diff
@@ -0,0 +1,198 @@
...
+        {
+            return;
+        }
+
+        if (!IsSystemLongPathEnabled())
+        {
+            throw new InvalidOperationException(
+                $"The path exceeds the Windows MAX_PATH limit of {MaxPath} characters and long path support is not enabled on this system. " +
+                "Visit https://aka.ms/enable-long-paths-on-windows for guidance on enabling long paths.
+                "  2. Use a shorter directory path.");
```
Comment (zateutsch) [inline src/winapp-CLI/WinApp.Cli/Helpers/LongPathHelper.cs]:
```suggestion
                "Visit https://aka.ms/enable-long-paths-on-windows for guidance on enabling long paths.");
```

File: src/winapp-CLI/WinApp.Cli/Helpers/LongPathHelper.cs [hunk 7]
```diff
@@ -0,0 +1,197 @@
...
+        if (path.Length <= MaxPath)
+        {
+            return;
+        }
+
+        if (!IsSystemLongPathEnabled())
+        {
+            throw new InvalidOperationException(
+                $"The path exceeds the Windows MAX_PATH limit of {MaxPath} characters and long path support is not enabled on this system. " +
+                "Visit https://aka.ms/enable-long-paths-on-windows for guidance on enabling long paths.");
```
Comment (zateutsch) [inline src/winapp-CLI/WinApp.Cli/Helpers/LongPathHelper.cs]:
```suggestion
                $"The path exceeds the Windows MAX_PATH limit of {MaxPath} characters and long path support is not enabled on this system. Visit https://aka.ms/enable-long-paths-on-windows for guidance on enabling long paths.");
```

**Review by nmetulev — APPROVED** [review]
